### PR TITLE
Containerize mkv mr creation for release

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
Containerize mkv mr creation for release

## Verification Steps
comment out commands in `build_deploy.sh` as in first commit and run:

```
AUTHOR_EMAIL=<from vault> AUTHOR_NAME=<from vault> GITLAB_TOKEN=<from vault> ./build_deploy.sh
```

confirm that MR is created in mkv repo (see example: https://gitlab.cee.redhat.com/mk-ci-cd/managed-kafka-versions/-/merge_requests/81)

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- ~~[ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
- ~~[ ] Documentation added for the feature~~
- ~~[ ] CI and all relevant tests are passing~~
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: create_release.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  yq -i ".service.image.tag = \"$VERSION\"" services/kas-fleet-manager.yaml
+  git config user.name "${AUTHOR_NAME}"
+  git config user.email "${AUTHOR_EMAIL}"
+  git commit -a -m "kas-fleet-manager stage release $VERSION"
+  # create an MR in managed-kafka-versions repo
+  echo "Creating MR with new config in managed-kafka-versions repository"
+  git push --force -o merge_request.create="true" -o merge_request.title="kas-fleet-manager stage release $VERSION" -o merge_request.description="https://gitlab.cee.redhat.com/service/kas-fleet-manager/-/compare/$CURRENT_COMMIT_SHA...$LATEST_COMMIT" -o merge_request.merge_when_pipeline_succeeds="false" -u origin "$BRANCH_NAME"
+else
+  echo "No new version detected for kas-fleet-manager"
+fi
```
Comment (machi1990) [inline create_release.sh:25]:
```suggestion
fi

```

File: docker/Dockerfile_build [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.6
+
+RUN curl -Lo /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v4.18.1/yq_linux_amd64 && \
+    chmod +x /usr/local/bin/yq
+
+RUN microdnf install git
+
+COPY . /src/
```
[the remark this answers is not in the document]
Comment (pawelpaszki) [reply docker/Dockerfile_build:8]: well. not exactly. I need the commit sha, but I could grab it earlier and pass as a param to the image. then I guess just copying the script is fine. will change it now

[the remark this answers is not in the document]
Comment (pawelpaszki) [reply docker/Dockerfile_build]: @machi1990 - addressed your comments and tested (see: https://gitlab.cee.redhat.com/mk-ci-cd/managed-kafka-versions/-/merge_requests/82)

**Review by machi1990 — COMMENTED** [review]
the change looks good to me.

I'd like @ziccardi or @daveconde to have a look as well

File: create_release.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+cd managed-kafka-versions
+BRANCH_NAME="kas-fleet-manager-${VERSION}"
+# only update the config, if different
+CURRENT_COMMIT_SHA=$(yq '.service.scm.commitSha' services/kas-fleet-manager.yaml)
+echo "Checking if the latest commit sha: $LATEST_COMMIT is different than current config commit sha: $CURRENT_COMMIT_SHA"
+if [[ "${CURRENT_COMMIT_SHA}" != "${LATEST_COMMIT}" ]]; then
+  git checkout -b "$BRANCH_NAME"
+  echo "Updating commit sha and image tag for kas-fleet-manager configuration"
+  # update commitSha
+  yq -i ".service.scm.commitSha = \"$LATEST_COMMIT\"" services/kas-fleet-manager.yaml
```
Comment (ziccardi) [inline create_release.sh:13]: Isn't this going to reformat the whole file?

Comment (pawelpaszki) [reply create_release.sh:13]: its not. see sample MRs in the comments/ description

Comment (ziccardi) [reply create_release.sh:13]: You are right. The formatting occurs only when array are used in the yaml, and this is not the case.

**Review by ziccardi — APPROVED** [review]
lgtm

**Pull request merged**